commercetools-models/src/main/java/io/sphere/sdk/discountcodes/commands/updateactions/SetCustomField.java

```diff
@@ -0,0 +1,34 @@
+package io.sphere.sdk.discountcodes.commands.updateactions;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.sphere.sdk.discountcodes.DiscountCode;
+import io.sphere.sdk.json.SphereJsonUtils;
+import io.sphere.sdk.types.customupdateactions.SetCustomFieldBase;
+
+/**
+ * Updates a custom field.
+ *
+ * {@doc.gen intro}
+ *
+ * {@include.example io.sphere.sdk.discountcodes.DiscountCodeCustomFieldsIntegrationTest#createDiscountCodeWithCustomType()}
+ *
+ * @see io.sphere.sdk.types.Custom
+ */
+public final class SetCustomField extends SetCustomFieldBase<DiscountCode> {
```

Comment:
in most classes that extends `SetCustomFieldBase` we add methods `ofJson`, `ofObject` and `ofUnset`. wouldn't it make sense to add it to `SetCustomFieldBase` 🤔  ?

Reply:
They're static and always use the specific update action. And since our update action are specific to the resource type they're updating I currently don't see a better design 😞 Only solution that I see is a code generator for this very specific case. But that sounds like overkill to me. What do you think?

Reply:
Yeah the code generator solution seems like an overkill and after looking up there doesen't seem simple workaround 😢 
I would agree to let it that way 👍